src/main/java/com/squareup/kotlinpoet/LambdaTypeName.kt

```diff
@@ -55,18 +55,33 @@ class LambdaTypeName internal constructor(
   }
 
   companion object {
-    /** Returns a lambda type with `returnType` and parameters of listed in `parameters`. */
+    /*
+     * Returns a lambda type with `returnType` and parameters listed in `parameters` and
+     * `namedParameters`.
+     */
     @JvmStatic fun get(
         receiver: TypeName? = null,
         parameters: List<TypeName> = emptyList(),
-        returnType: TypeName)
-        = LambdaTypeName(receiver, parameters, returnType)
+        namedParameters: List<Parameter> = emptyList(),
```

Comment:
It's weird to have two lists. I can't mix and match to get like `x: Long, y, z: Long`.

Reply:
Didn't wanna break existing API, and adding a new factory method with `parameters: List<Parameter>` wouldn't compile due to type erasure. If breaking is fine - I'd just get rid of `parameters: List<TypeName>`.

Reply:
Hmm we can use `@JvmName` to defeat the erasure 🙈

Reply:
Oh yeah, let me try that. Wonder how that would work for JS, no erasure - no problem?

Reply:
Well, I couldn't come up with a meaningful value for `@JvmName`, so I went the following way instead: the version with `parameters` and `namedParameters` stays, but is deprecated in favor of the version with `parameters: List<Parameter>`. We can remove the former when going 1.0. Does this make sense?

Reply:
I’m not too concerned with backwards-compatibility at this point.

Reply:
Agree. So would a single factory method with `parameters: List<ParameterSpec>` be enough, given that we'll have versions with `vararg parameters: ParameterSpec` and `vararg parameters: TypeName`?